### PR TITLE
Process Tekton Signature and Show "Signed" Badge

### PR DESCRIPTION
**Pull request by QuanZhang-William**
The [Tekton Trusted Resources](https://github.com/tektoncd/community/blob/main/teps/0091-trusted-resources.md#sign-the-resources) feature introduces digital signatures (stored in `tekton.dev/signature` annotation) for Tekton resources. This commit parses the new signature annotation and show a "Signed" badge for Tekton resources with such annotation.

Signed-off-by: Quan Zhang <zhangquan@google.com>

### CODE REVIEW

**Comment by QuanZhang-William** [comment]
cc @tegioz PTAL 😁

File: internal/tracker/source/tekton/tekton.go [hunk 1]
```diff
@@ -553,5 +554,20 @@ func enrichPackageFromAnnotations(p *hub.Package, annotations map[string]string)
 		}
 	}
 
+	// Sign key
+	if v, ok := annotations[signatureAnnotation]; ok {
+		var signature string
+		if err := yaml.Unmarshal([]byte(v), &signature); err != nil {
+			errs = multierror.Append(errs, fmt.Errorf("%w: invalid signature value", errInvalidAnnotation))
+		} else {
+			if signature == "" {
```
Comment (tegioz) [inline internal/tracker/source/tekton/tekton.go:563]: Can we be a bit more permissive for now and not require the signature to be present? If we return an error when the signature isn't there, the package won't be processed.

File: internal/tracker/source/tekton/tekton.go [hunk 2]
```diff
@@ -553,5 +554,20 @@ func enrichPackageFromAnnotations(p *hub.Package, annotations map[string]string)
...
+				errs = multierror.Append(errs, fmt.Errorf("%w: signature value not provided", errInvalidAnnotation))
+			} else {
+				p.Signatures = []string{signature}
```
Comment (tegioz) [inline internal/tracker/source/tekton/tekton.go:566]: We use the `signatures` field to track the kind of signatures found in the package, not the signature itself (please see signature in [containers images](https://github.com/artifacthub/hub/blob/b0a38faed699952c3202001693e5b933aac8a078/internal/tracker/source/container/container.go#L329) or in [Helm charts](https://github.com/artifacthub/hub/blob/b0a38faed699952c3202001693e5b933aac8a078/internal/tracker/source/helm/helm.go#L288-L315) as an example). Given this is a new kind of signature specific for Tekton, we could define a constant for it with the `tekton` value.

**Review by tegioz — CHANGES_REQUESTED** [review]
Hi @QuanZhang-William 👋 

Thanks for the PR!

File: internal/tracker/source/tekton/tekton.go (hunk 2 above)
Comment (QuanZhang-William) [reply internal/tracker/source/tekton/tekton.go]: Sorry for missing the point! Made the change as you suggested.

File: internal/tracker/source/tekton/tekton.go (hunk 1 above)
Comment (QuanZhang-William) [reply internal/tracker/source/tekton/tekton.go]: Ahh, good point! I made some modifications, PTAL.

**Review by tegioz — APPROVED** [review]
Thanks @QuanZhang-William! 👍

**Pull request merged**